integration/ethereummock/tx_handler.go

```diff
@@ -1,26 +1,90 @@
 package ethereummock
 
 import (
+	"bytes"
+	"encoding/gob"
+
+	"github.com/obscuronet/obscuro-playground/go/ethclient/txhandler"
+
 	"github.com/ethereum/go-ethereum/common"
 	"github.com/ethereum/go-ethereum/core/types"
-	"github.com/obscuronet/obscuro-playground/go/ethclient/mgmtcontractlib"
 	"github.com/obscuronet/obscuro-playground/go/obscurocommon"
 )
 
+const (
+	depositTxID uint64 = iota
+	rollupTxID
+	storeSecretTxID
+	requestSecretTxID
+)
+
 // MockTxHandler implements mgmtcontractlib.TxHandler for the ethereummock package
-// it never PackTx (because the mock eth takes care of it)
-// it always UnPackTx given the expected direct conversion of types.Transaction.Data -> obscurocommon.L1TxData
+// PackTx encodes the obscurocommon.L1Transaction in the Data field of the types.LegacyTx
+// and specifies the obscurocommon.L1Transaction type in the Gas Field (since it is not used for anything else)
+//
+// UnPackTx does the reverse steps - understands obscurocommon.L1Transaction to return based on the types.Transaction Gas
+// and decodes the correct object
 type MockTxHandler struct{}
 
-func (m MockTxHandler) PackTx(*obscurocommon.L1TxData, common.Address, uint64) (types.TxData, error) {
-	panic("implement me")
+func (m *MockTxHandler) PackTx(tx obscurocommon.L1Transaction, _ common.Address, nonce uint64) (types.TxData, error) {
+	var buf bytes.Buffer
+	enc := gob.NewEncoder(&buf)
+
+	if err := enc.Encode(tx); err != nil {
+		panic(err)
+	}
+
+	gasType := uint64(0)
+
+	switch tx.(type) {
+	case *obscurocommon.L1RollupTx:
+		gasType = rollupTxID
+	case *obscurocommon.L1DepositTx:
+		gasType = depositTxID
+	case *obscurocommon.L1RequestSecretTx:
+		gasType = requestSecretTxID
+	case *obscurocommon.L1StoreSecretTx:
+		gasType = storeSecretTxID
+	}
+
+	return &types.LegacyTx{
+		Gas:   gasType,
+		Nonce: nonce,
+		Data:  buf.Bytes(),
+	}, nil
 }
 
-func (m MockTxHandler) UnPackTx(tx *types.Transaction) *obscurocommon.L1TxData {
-	t := obscurocommon.TxData(tx)
-	return &t
+func (m *MockTxHandler) UnPackTx(tx *types.Transaction) obscurocommon.L1Transaction {
+	if len(tx.Data()) == 0 {
+		panic("Data cannot be 0 in the mock implementation")
+	}
+
+	// prepare byte buffer
+	buf := bytes.NewBuffer(tx.Data())
+	dec := gob.NewDecoder(buf)
+
+	// check type
+	var t obscurocommon.L1Transaction
+	switch tx.Gas() {
```

Comment:
this is super weird. We need a better way

Reply:
I've updated the comments and changed from using the Gas to using the To field.

Using an unused field in the types.transaction seems like a good choice, but I agree that it is weird.

Given that the ethereum mock doesnt execute contracts we need a way to distinguish what operation is being executed, I used a field in the transactions so not to break compliance with eth, but maybe there's a better way ?

Will take sugestions :)